*** phamm-0.5.18/lib/xhtml.php	2009-05-25 10:43:21.000000000 +0200
--- net-perversia__phamm/lib/xhtml.php	2010-04-12 17:31:51.000000000 +0200
***************
*** 675,684 ****
      $tag .= '<td colspan="2"><b>'.$domain_new.'</b></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" name="password1" value="" size="15" maxlength="15" /></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Confirm Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" name="password2" value="" size="15" maxlength="15" /></td></tr>'."\n";
      
      foreach ($pv as $name => $p_value)
      {
--- 675,687 ----
      $tag .= '<td colspan="2"><b>'.$domain_new.'</b></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" id="password1" name="password1" value="" size="15" maxlength="15" /></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Confirm Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" id="password2" name="password2" value="" size="15" maxlength="15" /></td></tr>'."\n";
! 
!     $tag .= '<tr><td><input type="button" id="button_generate_password" value="Generate Password" onclick="suggestPassword()" /></td>'."\n";
!     $tag .= '<td colspan="2"><input type="text" name="generated_pw" id="generated_pw" /></td></tr>' . "\n";
      
      foreach ($pv as $name => $p_value)
      {
***************
*** 764,773 ****
      $tag .= '<td colspan="2"><b>'.$account_new.'</b></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" name="password1" value="" size="15" maxlength="15" /></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Confirm Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" name="password2" value="" size="15" maxlength="15" /></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Name").'</td>'."\n";
      $tag .= '<td colspan="2"><input type="text" name="givenname" value="'.stripslashes($givenName).'" size="15" maxlength="50" /></td></tr>'."\n";
--- 767,779 ----
      $tag .= '<td colspan="2"><b>'.$account_new.'</b></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" id="password1" name="password1" value="" size="15" maxlength="15" /></td></tr>'."\n";
  
      $tag .= '<tr><td>'._("Confirm Password").'</td>'."\n";
!     $tag .= '<td colspan="2"><input type="password" id="password2" name="password2" value="" size="15" maxlength="15" /></td></tr>'."\n";
! 
!     $tag .= '<tr><td><input type="button" id="button_generate_password" value="Generate" onclick="suggestPassword()" /></td>'."\n";
!     $tag .= '<td colspan="2"><input type="text" name="generated_pw" id="generated_pw" /></td></tr>' . "\n";
  
      $tag .= '<tr><td>'._("Name").'</td>'."\n";
      $tag .= '<td colspan="2"><input type="text" name="givenname" value="'.stripslashes($givenName).'" size="15" maxlength="50" /></td></tr>'."\n";
*** phamm-0.5.18/www-data/functions.js	2009-05-25 10:48:50.000000000 +0200
--- net-perversia__phamm/www-data/functions.js	2010-04-12 17:11:05.000000000 +0200
***************
*** 36,38 ****
--- 36,54 ----
      f.setAttribute("autocomplete", "off");
      u.focus();
  }
+ function suggestPassword() {
+     // restrict the password to just letters and numbers to avoid problems:
+     // "editors and viewers regard the password as multiple words and
+     // things like double click no longer work"
+     var pwchars = "abcdefhjmnpqrstuvwxyz23456789ABCDEFGHJKLMNPQRSTUVWYXZ";
+     var passwordlength = 15;    // do we want that to be dynamic?  no, keep it simple :)
+     var passwd = document.getElementById('generated_pw');
+     passwd.value = '';
+ 
+     for ( i = 0; i < passwordlength; i++ ) {
+         passwd.value += pwchars.charAt( Math.floor( Math.random() * pwchars.length ) )
+     }
+     document.getElementById('password1').value = passwd.value;
+     document.getElementById('password2').value = passwd.value;
+     return true;
+ }
*** phamm-0.5.18/www-data/style/phamm/main.css	2009-05-25 10:49:56.000000000 +0200
--- net-perversia__phamm/www-data/style/phamm/main.css	2010-04-12 17:28:14.000000000 +0200
***************
*** 100,105 ****
--- 100,120 ----
      background-color: #f2ffb8;
  }
  
+ input[type="button"]
+ {
+     border-style: outset;
+     border-width: 1px;
+     color: #42516c;
+     background-color: #ffffff;
+     font-size: 1em;
+ }
+ 
+ input[type="button"]:hover
+ {
+     color: #450604;
+     background-color: #f2ffb8;
+ }
+ 
  input:focus
  {
      background-color: #ffffaa;
